### PR TITLE
[t130202] MIDE DEV V15 Migration - Operating Unit (OCA & BT) - Part 2

### PR DESCRIPTION
**Pull request by BT-apadron**
Fix method prepare_purchase_order

 
<div> Links to Odoo: </div> <ul>
<li><a target="_blank" href="https://odoo.braintec.com/web#view_type=form&model=project.task&id=130202">[t130202] MIDE DEV V15 Migration - Operating Unit (OCA & BT) - Part 2</a></li>
</ul>
<div>Affected Modules:</div>
<table><thead><tr><th>Module</th><th>Ext</th></tr></thead>
<tr><td>purchase_operating_unit</td><td>.py</td></tr>
</tbody></table>

### CODE REVIEW

File: purchase_operating_unit/models/stock_rule.py [hunk 1]
```diff
@@ -5,16 +5,12 @@
...
 
-    def _prepare_purchase_order(
-        self, product_id, product_qty, product_uom, origin, values, partner
-    ):
-        res = super(StockRule, self)._prepare_purchase_order(
-            product_id, product_qty, product_uom, origin, values, partner
-        )
-        if origin and "SO" in origin:
+    def _prepare_purchase_order(self, company_id, origins, values):
+        res = super()._prepare_purchase_order(self, company_id, origins, values)
```
Comment (BT-jmontero) [inline purchase_operating_unit/models/stock_rule.py:9]: self is not to be explicitly passed as an arg

File: purchase_operating_unit/models/stock_rule.py [hunk 2]
```diff
@@ -5,16 +5,12 @@
...
+        if origins and "SO" in origins:
```
Comment (BT-jmontero) [inline purchase_operating_unit/models/stock_rule.py:10]: origins sounds plural. Could it be a list of origins? If that is the case, then the code needs to be updated accordingly. In v13, origin was supposed to be one string, but if now it is supposed to be a list of strings... Please double check and debug to see what you receive.

**Review by BT-jmontero — COMMENTED** [review]
@BT-apadron Please check.

File: purchase_operating_unit/models/stock_rule.py (hunk 2 above)
Comment (BT-apadron) [reply purchase_operating_unit/models/stock_rule.py:10]: Solved in PR https://github.com/brain-tec/operating-unit/pull/350

**Pull request merged**